File: src/tugas.php
<?php
/* Standard Header */
require 'utilities/db.php';
require 'utilities/model.php';
require 'utilities/view.php';
require 'models/tugas.php';

session_set_cookie_params(30 * 24 * 60 * 60); 
session_start();
if (!isset($_SESSION["user"]))
{
	// user sudah login, dialihkan ke halaman lain
	header('Location: index.php');
}
else
{
	if (isset($_GET["id"]))
	{
		$tugas = new Tugas();
		$data = $tugas->getTugas($_GET["id"]);
		
		if ($data)
		{			
			$view = new View('views/tugas/default.tpl');
			$view->set('title', 'Todolist | Rincian Tugas');
			$view->set('headTags', '<script src="./scripts/tugas.js" type="application/javascript"></script><script type="text/javascript" src="./scripts/datetimepicker.js"></script><link rel="stylesheet" type="text/css" href="styles/tugas.css" />');
			
			echo json_encode($data);
			// $view->set('namaTugas', $data['nama']);
			// $view->set('deadline', date($data['tgl_deadline']));
			// $view->set('tags', $data['tag']);
			
			// $tagsLen = count($data['tag']);
			// $tagsEdit = '';
			// if ($tagsLen > 0)
			// {
			// 	$tagsEdit .= $data['tag'][0];
			// 	for ($i=1; $i < $tagsLen; $i++)
			// 	{
			// 		$tagsEdit .= ', ' . $data['tag'][$i];
			// 	}
			// }
			// $view->set('tagsEdit', $tagsEdit);
			
			// echo $view->output();
		}
		else
		{
			$view = new View('views/tugas/error.tpl');
			$view->set('title', 'Todolist | Rincian Tugas');
			$view->set('headTags', '');
			echo $view->output();
		}
	}
	else
	{
		header('Location: dashboard.php');
	}
}

File: TodolistJava/src/java/id/ac/itb/todolist/controller/Tugas.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package id.ac.itb.todolist.controller;

import id.ac.itb.todolist.dao.TugasDao;
import id.ac.itb.todolist.model.User;
import java.io.IOException;
import java.io.PrintWriter;
import javax.servlet.RequestDispatcher;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;

/**
 *
 * @author Edward Samuel
 */
public class Tugas extends HttpServlet {

    /**
     * Processes requests for both HTTP
     * <code>GET</code> and
     * <code>POST</code> methods.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        response.setContentType("text/html;charset=UTF-8");
        PrintWriter out = response.getWriter();
        HttpSession session = request.getSession();
        
        User user = (User) session.getAttribute("user");
        if (user == null)
        {
            // user sudah login, dialihkan ke halaman lain
            response.sendRedirect("./");
        }
        else
        {
            String idTugasStr = request.getParameter("id");
            if (idTugasStr != null)
            {
                int idTugas = Integer.parseInt(idTugasStr);
                
                TugasDao tugasDao = new TugasDao();
                
                RequestDispatcher dispathcer;
                if (!tugasDao.isAvailable(idTugas))
                {
                    dispathcer = request.getRequestDispatcher("/WEB-INF/views/tugas/error.jsp");
                }
                else if (tugasDao.isPemilik(idTugas, user.getUsername()))
                {
                    dispathcer = request.getRequestDispatcher("/WEB-INF/views/tugas/creator.jsp");
                }
                else if (tugasDao.isAssignee(idTugas, user.getUsername()))
                {
                    dispathcer = request.getRequestDispatcher("/WEB-INF/views/tugas/assignee.jsp");
                }
                else
                {
                    dispathcer = request.getRequestDispatcher("/WEB-INF/views/tugas/default.jsp");
                }
                
                request.setAttribute("headTags", "<link rel=\"stylesheet\" type=\"text/css\" href=\"styles/tugas.css\" /><link rel=\"stylesheet\" type=\"text/css\" href=\"./styles/autosuggest.css\" /><script src=\"./scripts/tugas.js\" type=\"application/javascript\"></script><script type=\"text/javascript\" src=\"./scripts/autosuggest2.js\"></script><script type=\"text/javascript\" src=\"./scripts/assigneesSuggestion.js\"></script>");
                request.setAttribute("bodyAttrs", "onload=\"onload(" + idTugas + ");\"");
                request.setAttribute("title", "Todolist | Rincian Tugas");  
                dispathcer.forward(request, response);
            }
            else
            {
                response.sendRedirect("./dashboard.jsp");
            }
        }
    }

    // <editor-fold defaultstate="collapsed" desc="HttpServlet methods. Click on the + sign on the left to edit the code.">
    /**
     * Handles the HTTP
     * <code>GET</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        processRequest(request, response);
    }

    /**
     * Handles the HTTP
     * <code>POST</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        processRequest(request, response);
    }

    /**
     * Returns a short description of the servlet.
     *
     * @return a String containing servlet description
     */
    @Override
    public String getServletInfo() {
        return "Short description";
    }// </editor-fold>
}
